Create setup.py for ri-public-examples

jerryjliu opened the pull request:
This PR creates a setup.py file and a corresponding package `ri_public_examples` which contains a single file, download_files.py. Now you can import ri-public-examples as a module! e.g.

```
from ri_public_examples import download_files
download_files('tabular/fraud', 'fraud')
```

This will fetch from data_urls/model_urls from the remote folder in Github, and download it to an output directory. The user can then choose to pip install the repo directly from `pip install -e git+https://github.com/RobustIntelligence/ri-public-examples.git`

Commit:
create setup.py for ri-public-examples

jerryjliu commented:
If this looks good I will edit the PR https://github.com/RobustIntelligence/rime/pull/7287 accordingly. Also note that this breaks the upload script for uploading over ri-public-examples to a cluster, but I will edit that as well

Commit:
oops

hinthornw commented on setup.py:
```diff
@@ -0,0 +1,18 @@
+    description="Package to download datasets/models/configs for use in RIME",
+    long_description=long_description,
+    long_description_content_type="text/markdown",
+    # TODO(blaine): upgrade to semver 3 when it is released.
```
do we need this todo?

jerryjliu replied:
removed :)

jerryjliu commented on ri_public_examples/download_files.py:
```diff
@@ -0,0 +1,79 @@
+import os
```
this file is mostly copied over from before, i'm not sure why this is registering as a whole new diff. i'll highlight the main parts that have changed

jerryjliu commented on ri_public_examples/download_files.py:
```diff
@@ -0,0 +1,79 @@
+from pathlib import Path
+
+
+GITHUB_URL = "https://raw.githubusercontent.com/RobustIntelligence/ri-public-examples/master/"
```
this repo URL allows us to download the data_urls.csv/model_urls.csv given the relative paths for a project

hwchase17 replied:
if we are pip installing this package, do we want to just make the stuff this points to (eg the data_urls.csv) included in the package?

jerryjliu replied:
you mean as data files? not sure what the benefits of that are vs. just hosting the links here (we'd still have to download all the raw data/model files somehow), what are your thoughts?

hwchase17 replied:
wouldnt have to do the whole GITHUB_URL download and stuff (since it will already be in the package) so could just load from that file and not have to download

jerryjliu replied:
mm it's just a bit more annoying because i have to add a manifest file + use importlib ( http://sixty-north.com/blog/including-package-data-in-python-packages.html), but ok i can try

hwchase17 replied:
can also leave it as a todo?

hwchase17 replied:
i think the user facing interface should remain the same

hwchase17 replied:
so can just have a todo to change bbehind the scene? i think it is a bit nicer but not a blcoker

jerryjliu replied:
👍 i'll leave a TODO, thanks

jerryjliu commented on ri_public_examples/download_files.py:
```diff
@@ -0,0 +1,79 @@
+    """Fetch urls and output paths from text file."""
+    # first download remote file path
+    local_path = out_dir / Path(remote_file_path).name
+    subprocess.run(["wget", remote_file_path, "-O", local_path])
```
this fetches the data_urls.csv/model_urls.csv to a local path (we could also read the url file directly using the `requests` library but that would necessitate extra dependencies)

hwchase17 replied:
this does assume that wget is installed on os, which i dont think it is by default for mac. would be nice to make more platform independent

jerryjliu replied:
mmm ok. i can try adding the `requests` lib as a dependency to fetch all files

hwchase17 replied:
what is the wget library we install in notebooks

jerryjliu replied:
we just use wget directly lol, it works in Colab

hwchase17 replied:
haha yeah was a bit confused by that. why do we install then?

jerryjliu replied:
sorry wdym by "why do we install". in Colab we just use the wget command directly

hwchase17 replied:
oops by install i meant import. why do we do the `import wget`?

jerryjliu replied:
i'm still confused, where do we do `import wget` OH do you mean in the notebooks? ....good catch we should take those out. we don't need them

hwchase17 replied:
ya!

jerryjliu replied:
Updated to remove dependency on any subprocess calls

jerryjliu commented on ri_public_examples/download_files.py:
```diff
@@ -0,0 +1,79 @@
+            subprocess.run(["unzip", "-o", full_outpath, "-d", out_zip_dir])
+
+
+def download_files(project_dir: str, out_dir: Optional[str] = None) -> None:
```
note that the user can also specify an out_dir to download the files to a local directory

jerryjliu commented on ri_public_examples/download_files.py:
```diff
@@ -0,0 +1,79 @@
+            f"ri-public-examples github repo."
+        )
+
+    project_dir = os.path.join(GITHUB_URL, project_dir)
```
using os.path.join here since GITHUB_URL is a http link

hwchase17 commented on ri_public_examples/download_files.py:
```diff
@@ -0,0 +1,79 @@
+    _download_files(model_urls_and_outpaths, out_dir / "models")
+
+
+if __name__ == "__main__":
```
when will this be used?

jerryjliu replied:
this was used before as a direct interface to the file, but we can remove now

Commit:
rm main

Commit:
cr

Commit:
update unzip command

jerryjliu commented:
Addressed comments: added `requests` lib as a dep in `setup.py`, converted wget calls to use the `requests` lib.  Also converted the `unzip` command to use the zipfile python lib, so we removed subprocess entirely in this script.

Also left a TODO to move data files to within the package. 

Finally I included an argument to download the config_file from ri-public-examples as optional. This isn't needed in the demo notebooks, but will help for things like our upload script
